service/telemetry.go

```diff
@@ -225,15 +228,34 @@ func (tel *telemetryInitializer) initOpenCensus(cfg telemetry.Config, telAttrs m
 	return pe, nil
 }
 
-func (tel *telemetryInitializer) initOpenTelemetry() (http.Handler, error) {
+func (tel *telemetryInitializer) initOpenTelemetry(attrs map[string]string) (http.Handler, error) {
 	// Initialize the ocRegistry, still used by the process metrics.
 	tel.ocRegistry = ocmetric.NewRegistry()
 
+	var resAttrs []attribute.KeyValue
+	for k, v := range attrs {
+		resAttrs = append(resAttrs, attribute.String(k, v))
+	}
+
+	res, err := resource.New(context.TODO(), resource.WithAttributes(resAttrs...))
+	if err != nil {
+		return nil, fmt.Errorf("error creating otel resources: %w", err)
+	}
+
 	exporter := otelprom.New()
-	tel.mp = sdkmetric.NewMeterProvider(sdkmetric.WithReader(exporter))
+	tel.mp = sdkmetric.NewMeterProvider(
+		sdkmetric.WithResource(res),
+		sdkmetric.WithReader(exporter),
+	)
 
 	registry := prometheus.NewRegistry()
-	if err := registry.Register(exporter.Collector); err != nil {
+	promLabels := make(prometheus.Labels)
+	for k, v := range attrs {
+		promLabels[sanitizePrometheusKey(k)] = v
+	}
```

Comment:
Per your comment, please add a TODO with a link to a tracking bug, to remove this when support for using Resource attributes into Prometheus exporter is added.

Reply:
I've added a comment about this linking to the specs, there is no bug because that is the intended behavior as per [OpenTelemetry specs](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#resource-attributes-1) and [OpenMetrics specs](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#supporting-target-metadata-in-both-push-based-and-pull-based-systems).

Reply:
Sorry for the delay, I was thinking about this.

Here is my understanding:
* When exporting to prometheus and we have Resource we generate a target_info metric.
* If we monitor that with collector's `prometheusreceiver` that we recommend, then the target_info metric is added back as resource. When exported to other backends most likely they get to be labels or available somehow as labels.
* I believe then we don't need anymore these values as prometheus labels.

Does it make sense to you? Maybe we protect this by another `featuregate` (do not add resource attributes as labels, enabled by default).

Reply:
Yes, that makes sense. Do you think we should add this `featuregate` in this pr or just remove this constant labels for now and add it back later?

Reply:
lets remove them for the moment, since everything is protected by the big featuregate to enable otel

Reply:
Done.